graphium/cli/finetune_utils.py

```diff
@@ -70,9 +63,70 @@ def benchmark_tdc_admet_cli(save_dir, wandb, name, inclusive_filter):
         ret = {k: v.item() for k, v in ret.items()}
         results[n] = ret
 
+    # Save to the results_dir by default or to the Hydra output_dir if needed.
+    # This distinction is needed, because Hydra's output_dir cannot be remote.
+    save_dir = cfg["constants"].get("results_dir", HydraConfig.get()["runtime"]["output_dir"])
     fs.mkdir(save_dir, exist_ok=True)
     path = fs.join(save_dir, "results.yaml")
     logger.info(f"Saving results to {path}")
 
     with fsspec.open(path, "w") as f:
         yaml.dump(results, f)
+
+
+@finetune_app.command(name="fp")
+def get_fingerprints_from_model(
+    fingerprint_layer_spec: List[str],
+    pretrained_model: str,
+    save_destination: str,
+    overrides: Optional[List[str]] = typer.Option(None, "--override", "-o", help="Hydra overrides"),
+):
+    """Endpoint for getting fingerprints from a pretrained model."""
+
+    if overrides is None:
+        overrides = []
+
+    with initialize(version_base=None, config_path="../../expts/hydra-configs"):
+        cfg = compose(config_name="main", overrides=overrides)
+        cfg = OmegaConf.to_container(cfg, resolve=True)
+
+    ## == Instantiate all required objects from their respective configs ==
+
+    # Accelerator
+    cfg, accelerator_type = load_accelerator(cfg)
+
+    # Data-module
+    datamodule = load_datamodule(cfg, accelerator_type)
+    datamodule.prepare_data()
+
+    # The predict_dataloader() returns either predict or test, so we need to run both.
+    datamodule.setup("test")
+    datamodule.setup("predict")
+
+    # Model
+    model = PredictorModule.load_pretrained_model(
+        pretrained_model,
+        device=accelerator_type,
+    ).model
+
+    ## == Fingerprinter
+
+    fps = []
+    with Fingerprinter(network=model, fingerprint_spec=fingerprint_layer_spec) as fp:
+        for batch in tqdm.tqdm(datamodule.predict_dataloader(), desc="Getting fingerprints"):
+            # TODO (cwognum): This is a hack to make sure the features are float32 (instead of float16)
```

Comment:
What hack? I don't see a hack?

Reply:
I was referring to: 

```python
if isinstance(v, torch.Tensor) and torch.is_floating_point(v):
    batch["features"][k] = v.float()
```

This feels too hard-coded to me. Is it reasonable to always use this? E.g. does the model always have the `float32` dtype?

Reply:
You just have to make sure that, in the trainer, you use `precision=32`. Or is the problem here that you're not using the trainer, and it's unclear what's the dtype input?

I think that any dtype works due to this, called at the beginning of the `forward` https://github.com/datamol-io/graphium/blob/f96c1b60b4d86e2fde0f3660b2916c14de388749/graphium/trainer/predictor.py#L198

Reply:
Yeah, indeed, felt like a hack because I wasn't using the `predictor`. Changed this based on your comment! Thanks! 🙏